modules/patterns/pages/running-user-scripts-on-the-build-pipeline.adoc

```diff
@@ -0,0 +1,107 @@
+
+We can extend the build pipeline with the `run-script-oci-ta` task to integrate the execution of those tools for these use cases.
+
+== Prerequisites
+
+* You already have a component onboarded to {ProductName} using xref:building:using-trusted-artifacts.adoc[trusted artifacts] variants of the build pipeline.
+
+[IMPORTANT]
+====
+There is no non `oci-ta` variant of this task nor plans to introduce it. The prerequesitie listed above is a hard requisite
+====
+
+* The script to run and all its dependencies are part of the source git repository or the container image the task will use to run the script.
+
+* You are already familiar with the procedure to xref:building:customizing-the-build.adoc[customize the build pipeline].
+
+== Procedure
+
+=== Extend the build pipeline with an additional task between `prefetch-dependencies` and `build-images` tasks:
+
+[source, diff]
+----
+   pipelineSpec:
+     description: |
+       This pipeline is ideal for building multi-arch container images from a Containerfile while maintaining trust after pipeline customization.
+@@ -203,6 +203,33 @@ spec:
+         workspace: git-auth
+       - name: netrc
+         workspace: netrc
++    - name: run-script
++      params:
++        - name: ociStorage
++          value: $(params.output-image).script
++        - name: ociArtifactExpiresAfter
++          value: $(params.image-expires-after)
++        - name: SCRIPT_RUNNER_IMAGE <1>
++          value: quay.io/my-script-runner-image:latest@sha256:digest
++        - name: SCRIPT
++          value: my-script <2>
++        - name: HERMETIC
++          value: $(params.hermetic)
++        - name: SOURCE_ARTIFACT
++          value: $(tasks.prefetch-dependencies.results.SOURCE_ARTIFACT)
++      runAfter:
++      - prefetch-dependencies
++      taskRef:
++        params:
++        - name: name
++          value: run-script-oci-ta
++        - name: bundle
++          value: quay.io/konflux-ci/tekton-catalog/task-run-script-oci-ta:0.1@sha256:c0f627069353ebd6d1ed03c8657e281eaf11be63706ea38cc53caf16cf4ffd65
++        - name: kind
++          value: task
++        resolver: bundles
+----
+
+<1> Use `SCRIPT_RUNNER_IMAGE` parameter to specify the container image you want to use to run the script. The image must already ship all the dependencies and additional tools required to run the script.
+
+<2> Use `SCRIPT` parameter to specify the script to run. This can be a command using an absolute path on the container image (`make build`) , a command using the relative path of an script on the source repository (`./my-script.sh build`) or an inline `sh` script. Refer to link:https://github.com/konflux-ci/build-definitions/blob/main/task/run-script-oci-ta/0.1/README.md[the task documentation] for more examples.
```

Comment:
```suggestion
<2> Use `SCRIPT` parameter to specify the script to run. This can be a command using an absolute path on the container image (`make build`), a command using the relative path of a script on the source repository (`./my-script.sh build`) or an inline `sh` script. Refer to link:https://github.com/konflux-ci/build-definitions/blob/main/task/run-script-oci-ta/0.1/README.md[the task documentation] for more examples.
```